### PR TITLE
added cNMF recipe

### PR DESCRIPTION
**Pull request by dylkot**
Hi! Here I've added a recipe for my package cnmf (consensus non-negative matrix factorization) which is currently on pypi. The method was described in eLife https://elifesciences.org/articles/43803 and the code is on Github https://github.com/dylkot/cNMF. Thanks!

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Review by marcelm — COMMENTED** [review]
Thanks! I added a couple of comments. The CI linting failure was due to a temporary problem that has hopefully now been fixed. When you push your changes, CI will be re-triggered.

**Comment by dylkot** [comment]
Hi @marcelm - thanks for the comments! I thought I had fixed everything you noted but it still seems to not be working? So sorry about this but I find the error messages fairly uninterpretable so any guidance you have would be greatly appreciated!

**Comment by marcelm** [comment]
Tl; DR: Please add
```
  noarch: python
```
to the build section.

---

I don’t know if you’ve found it, but this is the relevant error (in the "lint" workflow):
```
17:27:18 BIOCONDA DEBUG Found: LintMessage(recipe=Recipe "cnmf", check=<class 'bioconda_utils.lint.check_noarch.should_be_noarch_generic'>, severity=<Severity.ERROR: 30>, title='The recipe should be build as `noarch`', body="\nPlease add:\n\n    build:\n      noarch: generic\n\nPackages that don't require a compiler to build are normally\narchitecture independent and go into the `noarch` subset of\npackages.", start_line=12, end_line=16, fname='recipes/cnmf/meta.yaml', canfix=True)
The following problems have been found:

ERROR: recipes/cnmf/meta.yaml:16: should_be_noarch_generic: The recipe should be build as `noarch`
Errors were found
```
This is the helpful part of the error message, unfortunately hidden in the weird "LintMessage" thing:

> Please add:
> ```
> build:
>  noarch: generic   # but see below, this should actually be noarch: python
> ```
> Packages that don't require a compiler to build are normally
> architecture independent and go into the `noarch` subset of
> packages

For packages that contain compiled code, the package needs to be built once for each supported platform (generally that would be macOS, Linux and Windows, except that Bioconda doesn’t build Windows packages).

To mark a package as platform independent (for example, because it only contains data), `noarch: generic` needs to be added. The package will then be built only once, but is installable on any platform.

For Python packages, there is a special `noarch: python` setting, which is similar to `noarch: generic` in that the package only needs to be built once, but then there’s some extra stuff done at install time that depends on the Python version (generating the `.pyc` files mostly, I believe).
